crates/vm/levm/src/precompiles.rs

```diff
@@ -0,0 +1,182 @@
+use bytes::Bytes;
+use ethrex_core::{Address, H160, U256};
+
+use crate::{
+    call_frame::CallFrame,
+    constants::{REVERT_FOR_RETURN, SUCCESS_FOR_RETURN},
+    errors::{InternalError, PrecompileError, VMError},
+};
+
+pub const ECRECOVER_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x01,
+]);
+pub const SHA2_256_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x02,
+]);
+pub const RIPEMD_160_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x03,
+]);
+pub const IDENTITY_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x04,
+]);
+pub const MODEXP_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x05,
+]);
+pub const ECADD_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x06,
+]);
+pub const ECMUL_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x07,
+]);
+pub const ECPAIRING_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x08,
+]);
+pub const BLAKE2F_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x09,
+]);
+pub const POINT_EVALUATION_ADDRESS: H160 = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x0a,
+]);
+
+pub const PRECOMPILES: [H160; 10] = [
+    ECRECOVER_ADDRESS,
+    SHA2_256_ADDRESS,
+    RIPEMD_160_ADDRESS,
+    IDENTITY_ADDRESS,
+    MODEXP_ADDRESS,
+    ECADD_ADDRESS,
+    ECMUL_ADDRESS,
+    ECPAIRING_ADDRESS,
+    BLAKE2F_ADDRESS,
+    POINT_EVALUATION_ADDRESS,
+];
+
+pub fn is_precompile(callee_address: &Address) -> bool {
+    PRECOMPILES.contains(callee_address)
+}
+
+pub fn execute_precompile(current_call_frame: &mut CallFrame) -> Result<(u8, Bytes), VMError> {
```

Comment:
The biggest change that I'd do would be to change the return type of this to:
`Result<Bytes, VMError>`
Then, on each precompile function (like `ecmul(..)`) we can use the `?` for propagating the error onto the upper layer.

Reply:
Done in [Change execute_precompile return values](https://github.com/lambdaclass/ethrex/pull/1454/commits/6cb7ade0bade8f3d5e478ad4ca6cc4c41296625a)